crates/bevy_sprite/src/render/mod.rs

```diff
@@ -304,26 +336,180 @@ impl Default for SpriteMeta {
     }
 }
 
-const QUAD_INDICES: [usize; 6] = [0, 2, 3, 0, 1, 2];
+/// Sprite quad triangles vertex indices
+const QUAD_INDICES: [usize; 6] = [
+    0, 2, 3, // Bottom left triangle
+    0, 1, 2, // Top right triangle
+];
 
+/// Base Quad vertices 2D positions
 const QUAD_VERTEX_POSITIONS: [Vec2; 4] = [
+    // Top left
     Vec2::new(-0.5, -0.5),
+    // Top right
     Vec2::new(0.5, -0.5),
+    // Bottom right
     Vec2::new(0.5, 0.5),
+    // Bottom left
     Vec2::new(-0.5, 0.5),
 ];
 
+/// Base Quad vertices UV coordinates
 const QUAD_UVS: [Vec2; 4] = [
+    // Top left
     Vec2::new(0., 1.),
+    // Top right
     Vec2::new(1., 1.),
+    // Bottom right
     Vec2::new(1., 0.),
+    // Bottom left
     Vec2::new(0., 0.),
 ];
 
-#[derive(Component, Eq, PartialEq, Copy, Clone)]
+/// Component defining a batch of sprites for the render world
+#[derive(Component)]
 pub struct SpriteBatch {
+    /// The [`SpriteMeta`] vertex data indices
+    range: Range<u32>,
+    /// The texture handle id
     image_handle_id: HandleId,
+    /// Defines if the `range` targets [`SpriteMeta::vertices`] or [`SpriteMeta::colored_vertices`]
     colored: bool,
+    /// Sort key of the batch
+    z_order: f32,
+}
+
+pub fn prepare_sprites(
+    mut commands: Commands,
+    render_device: Res<RenderDevice>,
+    render_queue: Res<RenderQueue>,
+    mut sprite_meta: ResMut<SpriteMeta>,
+    gpu_images: Res<RenderAssets<Image>>,
+    mut extracted_sprites: ResMut<ExtractedSprites>,
+) {
+    // Impossible handle id value
+    const NULL_HANDLE: HandleId = HandleId::Id(Uuid::nil(), u64::MAX);
+
+    // Clear the vertex buffers
+    sprite_meta.vertices.clear();
+    sprite_meta.colored_vertices.clear();
+
+    // Sort sprites by z for correct transparency and then by handle to improve batching
+    let sort = |a: &ExtractedSprite, b: &ExtractedSprite| match a
+        .transform
+        .translation
+        .z
+        .partial_cmp(&b.transform.translation.z)
+    {
+        Some(Ordering::Equal) | None => a.image_handle_id.cmp(&b.image_handle_id),
+        Some(other) => other,
+    };
+    extracted_sprites.sprites.sort_unstable_by(sort);
+    extracted_sprites.colored_sprites.sort_unstable_by(sort);
+
+    for (colored, sprites) in [
+        (false, &extracted_sprites.sprites),
+        (true, &extracted_sprites.colored_sprites),
```

Comment:
I was thinking that doing this would be wrong as previously all sprites whether colour or not were sorted together but I guess the batches will be split as part of the batching code that is run after the phases are sorted. It could create more work for the batching to do though. Maybe need to compare performance for something with a mix of colour and not colour sprites.

Reply:
I introduced this separation in https://github.com/bevyengine/bevy/pull/5247/commits/0a0d5c4254d24a34f463765b675ebfd0485bdec0, previously colored and non colored were merged together.
I can rollback the changes but the performance improvements would go back to 1FPS instead of 3FPS

Reply:
It seems maybe they aren't being split in the batching system... maybe. I'm speculating.